drivers/sensor/lps22hb/lps22hb.c

```diff
@@ -56,7 +56,8 @@ static inline void lps22hb_press_convert(struct sensor_value *val,
 	/* Pressure sensitivity is 4096 LSB/hPa */
 	/* Convert raw_val to val in kPa */
 	val->val1 = (raw_val >> 12) / 10;
-	val->val2 = ((s32_t)((raw_val) & 0x0FFF) * 1000L) >> 12;
+	val->val2 = (raw_val >> 12) % 10 * 1000 +
```

Comment:
Actually, the measurement unit seems to be off. The value is computed as val = val1 + val2 * 1000000. Since val is expressed in kPa, than val2 should be mPa.

This means that val2 should be computed something like: (((s32_t)((raw_val) & 0x0FFF) * 100000L) >> 12).